src/gwt/src/org/rstudio/studio/client/workbench/views/files/FilesPane.java

```diff
@@ -236,6 +237,32 @@ public void setPosition(int offsetWidth, int offsetHeight)
        });
    }
 
+   @Override
+   public void showDataImportFileChoice(FileSystemItem file, 
+                                  Command onEdit,
+                                  Command onImport)
+   {
+       final ToolbarPopupMenu menu = new ToolbarPopupMenu();
+       
+       String editLabel = AppCommand.formatMenuLabel(
+          commands_.renameFile().getImageResource(), "Open in Editor", null);
+       String importLabel = AppCommand.formatMenuLabel(
+          StandardIcons.INSTANCE.import_dataset(), 
+          "Import Dataset", 
+          null);
+       
+       menu.addItem(new MenuItem(editLabel, true, onEdit));
```

Comment:
For CSV and any other text based format we can definitely have an "Open in Editor" option but for the other formats we're going to end up opening them externally. I think the menu should reflect this. The easy thing to do is just to rename "Open in Editor" to "View File" then let the default handlers do their thing (it will either show in the editor or externally as appropriate). 

I also think you should add ellipses to the dataset command, e.g. "Import Dataset...")